### PR TITLE
Set overlay driver as default in application manifest

### PR DESCRIPTION
**Pull request by knisbet**
Use overlay as the default storage driver. I don't know if overlay2 would be a better default, or if overlay is more likely to support a wider range of systems.

Updates https://github.com/gravitational/gravity/issues/211
Updates https://github.com/gravitational/gravity.e/issues/3913

Requires backport to 5.3.x/5.4.x.

### CODE REVIEW

**Comment by klizhentas** [comment]
I would prefer overlay2 - it has less issues (e.g no hard links exhaustion)

**Comment by knisbet** [comment]
@klizhentas alright I switched it to overlay2

File: docs/5.x/pack.md [hunk 1]
```diff
@@ -525,7 +525,8 @@ systemOptions:
 
   # Docker section allows to customize docker
   docker:
-    # Storage backend used, supported: "devicemapper" (default), "overlay", "overlay2"
+    # Storage backend used, supported: "devicemapper", "overlay", "overlay2" (default)
+    # Note: devicemapper is no longer available from gravity 5.3.4 or later.
```
Comment (r0mant) [inline docs/5.x/pack.md]: I would then just remove the reference to it from here and above. IIRC we have a note about it on the System Requirements page.

**Comment by knisbet** [comment]
@r0mant I agree we should remove all related code at some point, I was just being a bit cautious, as if someone comes to us and really wants device mapper on a more recent release we might need to make it work. But then again, if that were to come up, we could try and back out the code removal.

**Pull request merged**